tests/e2e/README.md

```diff
@@ -0,0 +1,142 @@
+<!--
+SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# NVIDIA Container Toolkit – End‑to‑End (E2E) Test Suite
+
+---
+
+## 1  Scope & Goals
+This repository contains a **Ginkgo v2 / Gomega** test harness that exercises an
+NVIDIA Container Toolkit (CTK) installation on a **remote GPU‑enabled host** via
+SSH.  The suite validates that:
+
+1. CTK can be installed (or upgraded) head‑less (`INSTALL_CTK=true`).
+2. The specified **container image** runs successfully under `nvidia-container-runtime`.
+3. Errors and diagnostics are captured for post‑mortem analysis.
+
+The tests are intended for continuous‑integration pipelines, nightly
+compatibility runs, and pre‑release validation of new CTK builds.
+
+---
+
+## 2  Execution model
+* The framework **does not** spin up a Kubernetes cluster; it drives a single
+  host reachable over SSH.
+* All commands run in a Ginkgo‑managed context (`ctx`) so they abort cleanly on
+  timeout or Ctrl‑C.
+* Environment discovery happens once in `TestMain` → `getTestEnv()`; parameters
+  are therefore immutable for the duration of the run.
```

Comment:
This seems like the kind of thing that can get out of sync with the actual implementation.